tests/codebuild/run_integtest.sh

```diff
@@ -45,15 +58,16 @@ function cleanup {
         echo "need_setup_cluster is true, tearing down cluster we created."
         eksctl delete cluster --name "${cluster_name}" --region "${cluster_region}"
         # Delete the role associated with the cluster thats being deleted
-        aws iam detach-role-policy --role-name "${role_name}" --policy-arn arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
-        aws iam delete-role --role-name "${role_name}"
+        delete_generated_role "${role_name}"
+        delete_generated_role "${default_role_name}"
     else
         echo "need_setup_cluster is not true, will remove operator without deleting cluster"
         kustomize build "$path_to_installer/config/default" | kubectl delete -f -
         # Delete the namespaced operator. TODO: This can be cleaner if parameterized
         kustomize build "$path_to_installer/config/crd" | kubectl delete -f -
-        generate_namespace_operator_installer "${crd_namespace}"
-        kubectl delete -f temp_file.yaml    
+        generate_operator_installer_for_given_role ${crd_namespace} "config/installers/rolebasedcreds/namespaced" "${role_name}"
+        kubectl delete -f temp_file.yaml 
```

Comment:
Reading from above it doesn't quite tells me where did `temp_file.yaml` come from? So either change the function to take the file path or echo the output from that function and write it here in file in line 68.

Reply:
good point, i cleaned up this logic by limiting the use of the temp_file.yaml to the function where it is generated.